contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -662,181 +619,137 @@ contract LiquidityProtection is ILiquidityProtection, Utils, Owned, ReentrancyGu
+        Fraction memory loss = _impLoss(addSpotRate, removeAverageRate);
 
-        // withdraw the pool tokens from the wallet
-        _withdrawPoolTokens(poolToken, poolLiquidationAmount);
+        // deduct the position IL from the target amount
+        targetAmount = _deductIL(Math.max(reserveAmount, targetAmount), loss);
 
-        // remove liquidity
-        _removeLiquidity(poolToken, poolLiquidationAmount, reserveToken, IReserveToken(address(_networkToken)));
+        // get the pool deficit
+        Fraction memory poolDeficit = _poolDeficit(poolToken);
 
-        // migrate the received tokens
-        uint256 removedAmount = reserveToken.balanceOf(address(this));
-        uint256 value;
-        if (reserveToken.isNativeToken()) {
-            value = removedAmount;
-        } else {
-            IERC20(address(reserveToken)).safeApprove(address(_networkV3), removedAmount);
+        // no deficit
+        if (poolDeficit.n == 0) {
+            return (targetAmount, targetAmount);
         }
-        _networkV3.migrateLiquidity{ value: value }(
-            reserveToken,
-            msg.sender,
-            fullyProtectedAmount,
-            removedAmount,
-            originalAmount
-        );
 
-        // if the contract still holds network tokens, burn them
-        uint256 networkBalance = _networkToken.balanceOf(address(this));
-        if (networkBalance > 0) {
-            _burnNetworkTokens(poolToken, networkBalance);
-        }
+        // calculate the available liquidity portion
+        Fraction memory availablePortion = Fraction({ n: poolDeficit.d - poolDeficit.n, d: poolDeficit.d});
+
+        // return the amount the provider will receive for removing liquidity
+        // as well as the specific position value (before deficit reduction)
+        return (_mulDivF(targetAmount, availablePortion.n, availablePortion.d), targetAmount);
     }
 
     /**
-     * @dev returns the amount the provider will receive for removing liquidity
+     * @dev returns the pool deficit based on the total protected amount vs. total
+     * positions value, in PPM
      */
-    function _removeLiquidityTargetAmount(
-        IDSToken poolToken,
-        IReserveToken reserveToken,
-        uint256 poolAmount,
-        uint256 reserveAmount,
-        PackedRates memory packedRates,
-        uint256 addTimestamp,
-        uint256 removeTimestamp
-    ) internal view returns (uint256) {
-        // get the rate between the pool token and the reserve token
-        Fraction memory poolRate = _poolTokenRate(poolToken, reserveToken);
+    function poolDeficitPPM(IDSToken poolToken)
+        external
+        view
+        returns (uint256)
+    {
+        Fraction memory poolDeficit = _poolDeficit(poolToken);
 
-        // calculate the protection level
-        Fraction memory level = _protectionLevel(addTimestamp, removeTimestamp);
+        // no deficit
+        if (poolDeficit.n == 0) {
+            return 0;
```

Comment:
Already handled correctly in the final `return` statement.

Reply:
removed